fv3config/config/diag_table.py

```diff
@@ -0,0 +1,212 @@
+            return f'"{token}"'
+        else:
+            return str(token)
+
+    @staticmethod
+    def _parse_line(line: str) -> Sequence[Union[str, int]]:
+        token_strings = line.replace(" ", "").strip(",").split(",")
+        return list(map(DiagTable._str_to_token, token_strings))
+
+    @classmethod
+    def from_dict(cls, diag_table: dict):
+        files = [dacite.from_dict(DiagTableFile, f) for f in diag_table["files"]]
+        return cls(diag_table["name"], diag_table["base_time"], files)
+
+    @classmethod
+    def from_str(cls, diag_table: str):
+        """Initialize DiagTable class from Fortran string representation."""
+        lines = diag_table.split("\n")
+        lines = [line for line in lines if line and line[0] != "#"]
+        parsed_lines = list(map(cls._parse_line, lines[2:]))
+
+        file_lines = []
+        field_lines = {}
+
+        for tokens in parsed_lines:
+            if len(tokens) == 6:
+                # line corresponds to definition of a file
+                file_name = tokens[0]
+                file_lines.append(tokens)
+                field_lines[file_name] = []
+            elif len(tokens) == 8:
+                # line corresponds to definition of a field
+                file_name = tokens[3]
+                if file_name not in field_lines:
+                    raise ConfigError(
+                        "Files must be defined before they can be used by a field in "
+                        f"diag_table. {file_name} has not been defined yet."
+                    )
+                field_lines[file_name].append(tokens)
+            else:
+                logger.warning(
+                    f"Ignoring a line that could not be parsed in diag_table: {tokens}"
+                )
+
+        files = []
+        for file_tokens in file_lines:
+            file_name = file_tokens[0]
+            fields = []
+            for field_tokens in field_lines[file_name]:
+                fields.append(
+                    DiagTableField(
+                        module_name=field_tokens[0],
+                        field_name=field_tokens[1],
+                        output_name=field_tokens[2],
+                        time_sampling=field_tokens[4],
+                        reduction_method=field_tokens[5],
+                        regional_section=field_tokens[6],
+                        packing=field_tokens[7],
+                    )
+                )
+            files.append(
+                DiagTableFile(
+                    name=file_name,
+                    frequency=file_tokens[1],
+                    frequency_units=file_tokens[2],
+                    fields=fields,
+                    file_format=file_tokens[3],
+                    time_axis_units=file_tokens[4],
+                    time_axis_name=file_tokens[5],
+                )
+            )
+        name = lines[0]
+        base_time = cls._str_to_time(lines[1])
+        return cls(name, base_time, files)
```

Comment:
This function is very large and has some clear taxonomies in what it's doing. Could you separate it into some well-named functions? After reading a while I think I see that the function splits the lines into some representation of file and field lines, and then for each of the files constructs a file object, the list of which can then be used to construct the class. If you put this in a small number of lines, I'd be more confident that's what it's doing (and I'd reach that conclusion more quickly).